### ISSUE TITLE
ERR_TIMED_OUT on multi requests

### ISSUE THREAD

**Issue by dizcza** [johann-petrak/python-tiny-http-server#7]
I'm using your tiny http server to serve firmware updates for my ESP32 boards.
I have ~10 such boards in total all requesting to download a file. HTTPS + auth mode. And as soon as they all start a request, the server is no longer reachable (even from my PC).

Also, I noticed that if any such request fails, the server is stalled and not recovered until reset.

Perhaps, threads + timeouts on error would help.

**Comment by dizcza** [comment]
The connection is weak causing more errors on both sides: server and client.

**Comment by johann-petrak** [comment]
Hi - the reason why this module is called "tiny" is because it avoids all the complexity of serving many connections in parallel, keeping long/slow connections open etc. 
If you need that please consider using a full-fledged server instead which supports all this. 

My feeling would be to close this as WONTFIX, but I will leave it open for a while if anyone wants to contribute a small and easy fix if it exists, but this should definitely not become a reinvention of much bigger wheels :)

Heavy duty is probably best achieved by using e.g. flask with gunicorn.

**Comment by dizcza** [comment]
I agree, feel free to close it as wontfix.

I've done some projects with flask and gunicorn and I'm also thinking about these tools but I'd rather pip install something ready to use :) I cannot believe that such a tool that graciously handles many parallel requests without crashing does not exist in python - I'm talking about HTTPS + auth server. But I cannot find anything! Perhaps you came across such tools?

**Comment by dizcza** [comment]
All right, I've come up with the simplest file-sending server on flask + gunicorn. Posting it here for other users:

```python
import os

from flask import Flask, send_file
from flask_httpauth import HTTPBasicAuth
from werkzeug.security import generate_password_hash, check_password_hash

app = Flask(__name__)
auth = HTTPBasicAuth()

FIRMWARE_PATH = "firmware.bin"

users = {
    os.environ['ENV_OTA_SERVER_USER']: generate_password_hash(os.environ['ENV_OTA_SERVER_PASS'])
}


@app.route(f'/{FIRMWARE_PATH}')
@auth.login_required
def download_firmware():
    return send_file(FIRMWARE_PATH, as_attachment=False)


@auth.verify_password
def verify_password(username, password):
    if username in users and \
            check_password_hash(users.get(username), password):
        return username


if __name__ == '__main__':
    app.run(debug=False, port=8070, host="0.0.0.0")
```

The script servers only one predefined file. And I'm launching it with gunicorn like so:

```
gunicorn --certfile ca_cert.pem --keyfile ca_key.pem --worker-class gevent -b 0.0.0.0:8070 app:app
```
